src/neptune/internal/init/run.py

```diff
@@ -343,12 +351,24 @@ def init_run(
     if mode != Mode.READ_ONLY:
         if name is not None:
             _run[attr_consts.SYSTEM_NAME_ATTRIBUTE_PATH] = name
+
         if description is not None:
             _run[attr_consts.SYSTEM_DESCRIPTION_ATTRIBUTE_PATH] = description
+
         if hostname is not None:
-            _run[attr_consts.SYSTEM_HOSTNAME_ATTRIBUTE_PATH] = hostname
+            _run[f"{monitoring_namespace}/hostname"] = hostname
+            if with_id is None:
+                _run[attr_consts.SYSTEM_HOSTNAME_ATTRIBUTE_PATH] = hostname
+
+        if pid is not None:
+            _run[f"{monitoring_namespace}/pid"] = str(pid)
```

Comment:
It was not specified in spec if it should be int or string.

Reply:
Yes but on Windows this could exceed our int range.

Reply:
Very unfortunate. :(